torch/csrc/jit/passes/batch_mm.cpp

```diff
@@ -77,10 +77,23 @@ static constexpr size_t min_fusion_size = 4;
 
 bool have_same_shape(at::TensorList inputs) {
   auto expected_sizes = inputs[0].sizes();
-  return std::all_of(
+  auto expected_strides = inputs[0].strides();
+  return (std::all_of(
       inputs.begin(), inputs.end(), [expected_sizes](const at::Tensor& t) {
         return t.sizes() == expected_sizes;
-      });
+      }) && std::all_of(
+      inputs.begin(), inputs.end(), [expected_strides](const at::Tensor& t) {
+        return t.strides() == expected_strides;
+      })) ;
```

Comment:
This requires a change of the function name, because you have completely change the semantics. Is it really slower if the strides are not all the same, or is it just a guess?

Reply:
It's so that in the transpose check I could check the strides of only the first tensor, and honestly it's hard to imagine graph that would have tensors of same sizes eligible for tree reduction, but with the different strides.  In any case, I'll just make transpose check go over all the tensors and leave `have_same_shape` alone.